### ISSUE TITLE
Do you need Terraria resources for your game?

### ISSUE THREAD

**Issue by egordorichev** [hydren/terrarium#2]
Hello! Some time ago I found Terraria resources such as Music, Testures, Sounds.  Maybe you need some? Best wishes, Egor.

**Comment by hydren** [comment]
Thanks, Egor! Yes, I need some resources. Most of the resources I garthered were made on Paint by a friend of mine. These resources you found are ripped from Terraria?

**Comment by egordorichev** [comment]
I just decompiled Terraria and found `content folder`. Using https://github.com/Antag99/TExtract you can convert `.xnb` to `.png` and `.wav` formats.

**Comment by egordorichev** [comment]
I moved this repo to github https://github.com/egordorichev/Terraria-Resources.

**Comment by hydren** [comment]
Thank you! It will be very useful!

**Comment by akien-mga** [comment]
Do you have the authorization of the Terraria developers for decompiling their soft and sharing the resources on open source repositories like BitBucket and GitHub? As far as I know these are copyrighted and proprietary assets.

**Comment by hydren** [comment]
I thought that you only needed authorization if we were to use copyrighted assets. Right now it is interesting to study their solution to spriting. Is decompilation-only (not using in the game) wrong?

**Comment by akien-mga** [comment]
> I thought that you only needed authorization if we were to use copyrighted assets. Right now it is interesting to study their solution to spriting. Is decompilation-only (not using in the game) wrong?

**Disclaimer: I am not a lawyer, so I might be mistaken in what follow :-)**

In my understanding, reusing copyrighted assets in another project is indeed no-go, unless you get explicit authorization from the copyright owners.

I don't know if decompiling assets to study them is OK, but I would say that as long as you do it for your own usage, there shouldn't be any problem.

What might be more problematic on the other hand is redistributing the decompiled assets as @egordorichev was doing: free redistribution is one of the values that free software grants us, so I guess it means that proprietary stuff can't be freely shared with others.

So basically, I'd say feel free to be in touch with @egordorichev to get a copy of those assets while working on the engine, but don't share them on any public repo :)

Hope this helps :)

**Comment by hydren** [comment]
Yes, thank you. The idea is to keep copyrighted code out of the repo. What I was intending to do is to take a look at the decompiled assets and study them. When we need an asset to upload to be used in the game, we would create one. Something like Freedoom project made with Doom.

**Comment by hydren** [comment]
However, I think Egor shouldn't have any trouble with the repo's. There is no actual game in his repos. Hardly any terraria developer would care about it.
